Automattic/wp-calypso
# Emails: Make the ThankYou react component to be a completely isolated component

#### Changes proposed in this Pull Request

Change a few components within the ThankYou component to be independent of the MarketPlace component.

#### Testing instructions

The test to perform for this change is the same as we had to perform in this [Pull Request](https://github.com/Automattic/wp-calypso/pull/54577):

```
There are two things we need to test:

1. The original Yoast purchase flow, and
2. The visuals for the new Professional Email page
```

The link to check the ThankYou page for the step 2 is: `/:domain/titan/setup-mailbox/thank-you/:slug`

Related to # {1200182182542585-as-1200182182542585}
